cpp/src/arrow/acero/query_context.cc

```diff
@@ -23,10 +23,19 @@ namespace arrow {
 using arrow::internal::CpuInfo;
 namespace acero {
 
+namespace {
+io::IOContext GetIoContext(QueryOptions& opts, const ExecContext& exec_context) {
```

Comment:
```suggestion
io::IOContext GetIoContext(const QueryOptions& opts, const ExecContext& exec_context) {
```

Reply:
I think exec_context can be make as constant, but custom_io_executor should be non-const, so QueryOptions cannot have const. Should I use `mutable` for `io::Executor custom_io_executor` or some other methods?